tests/components/modbus/test_sensor.py

```diff
@@ -201,7 +202,7 @@ async def test_config_sensor(hass, mock_modbus):
                 CONF_STRUCTURE: "2s",
                 CONF_SWAP: CONF_SWAP_WORD,
             },
-            "Error in sensor test_sensor swap(word) not possible due to the registers count: 1, needed: 2",
+            "Error in sensor test_sensor swap(word) not possible due to the registers size: 2, needed: 4 bytes",
```

Comment:
But normally register size is 2 (according to standard), when register_size is omitted, so it is kind of hard to understand the error message.

Apart from that count is valid in combination with register_size, so you could have e.g.
register_size: 4, count 2
Register_size: 8, count 1
Count: 4
And they all have the same size.